ci-operator/templates/openshift/installer/cluster-disaster-recovery-control-plane.yaml

```diff
@@ -0,0 +1,501 @@
+        wait "$!"
+
+    # Once the cluster is up, executes shared tests
+    - name: test
+      image: ${IMAGE_TESTS}
+      terminationMessagePolicy: FallbackToLogsOnError
+      resources:
+        requests:
+          cpu: 1
+          memory: 300Mi
+        limits:
+          memory: 3Gi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-profile
+        mountPath: /tmp/cluster
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      env:
+      - name: AWS_SHARED_CREDENTIALS_FILE
+        value: /tmp/cluster/.awscred
+      - name: AWS_DEFAULT_REGION
+        value: us-east-1
+      - name: ARTIFACT_DIR
+        value: /tmp/artifacts
+      - name: SSH_SECRET_KEY_PATH
+        value: /etc/openshift-installer/ssh-privatekey
+      - name: HOME
+        value: /tmp/home
+      - name: KUBECONFIG
+        value: /tmp/artifacts/installer/auth/kubeconfig
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        set -euo pipefail
+
+        export PATH=/usr/libexec/origin:$PATH
+
+        trap 'touch /tmp/shared/exit' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        mkdir -p "${HOME}"
+
+        # wait for the API to come up
+        while true; do
+          if [[ -f /tmp/shared/exit ]]; then
+            echo "Another process exited" 2>&1
+            exit 1
+          fi
+          if [[ ! -f /tmp/shared/setup-success ]]; then
+            sleep 15 & wait
+            continue
+          fi
+          # don't let clients impact the global kubeconfig
+          cp "${KUBECONFIG}" /tmp/admin.kubeconfig
+          export KUBECONFIG=/tmp/admin.kubeconfig
+          break
+        done
+
+        mkdir ~/.ssh
+        cp "${SSH_SECRET_KEY_PATH}" ~/.ssh/id_rsa
+        chmod 0600 ~/.ssh/id_rsa
+
+        echo "Setting up ssh bastion"
+        curl https://raw.githubusercontent.com/eparis/ssh-bastion/master/deploy/deploy.sh | bash
+        BASTION_HOST=$(oc get service -n openshift-ssh-bastion ssh-bastion -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+
+        # Machine API won't let the user to destroy the node which runs the controller
+        echo "Destroying two masters"
+        MAPI_POD=$(oc get pod -l k8s-app=controller -n openshift-machine-api --no-headers -o name)
+        SURVIVING_MASTER_NODE=$(oc get ${MAPI_POD} -n openshift-machine-api -o jsonpath='{.spec.nodeName}')
+
+        mapfile -t MASTER_NODES_TO_REMOVE < <(oc get nodes -l node-role.kubernetes.io/master= -o name | grep -v "${SURVIVING_MASTER_NODE}")
+        MASTER_MACHINES_TO_REMOVE=()
+        for master in ${MASTER_NODES_TO_REMOVE[@]}
+        do
+          MASTER_MACHINES_TO_REMOVE+=($(oc get ${master} -o jsonpath='{.metadata.annotations.machine\.openshift\.io\/machine}' | cut -d '/' -f 2))
+        done
+
+        echo "Destroy two masters"
+        oc --request-timeout=5s -n openshift-machine-api delete machines ${MASTER_MACHINES_TO_REMOVE[@]}
```

Comment:
One way to address this without a new template would be to drop all of this new stuff into a shell function, and jobs that wanted to trigger it could use:

```yaml
- name: TEST_COMMAND
  value: |
    kill-some-masters
    TEST_SUITE=openshift/conformance/parallel run-tests
```

Reply:
When you say drop this stuff into a new shell function, are you saying the new shell function would be added to an already existing template, like cluster-launch-installer-e2e.yaml or another template? That makes sense, but I wanted to make sure that is your point.

Reply:
> When you say drop this stuff into a new shell function, are you saying the new shell function would be added to an already existing template, like cluster-launch-installer-e2e.yaml or another template?

Yup, right [here][1], with:

```sh
function destroy-some-control-plane-machines() {
  ...bla bla...
}
```

[1]: https://github.com/openshift/release/blob/edabb5ebc4d5cb6f82ddefc18e775c2b0a8e2412/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L173

Reply:
Created #3572 to play with this idea